apps/server/src/services/rundown-service/RundownService.ts

```diff
@@ -301,3 +301,30 @@ export async function swapEvents(from: string, to: string) {
 function updateChangeNumEvents() {
   eventLoader.updateNumEvents();
 }
+
+/**
+ * Updates a property of the event with the given id
+ * @param {string} eventId
+ * @param {keyof OntimeEvent} propertyName
+ * @param {OntimeEvent[typeof propertyName]} propertValue
+ */
+export function updateEvent(
+  eventId: string,
+  propertyName: keyof OntimeEvent,
+  newValue: OntimeEvent[typeof propertyName],
+) {
+  const event = EventLoader.getEventWithId(eventId);
+
+  if (event) {
+    const propertiesToUpdate = { [propertyName]: newValue };
+
+    // Handles the special case for duration
+    if (propertyName === 'duration') {
+      propertiesToUpdate.timeEnd = event.timeStart + (newValue as number);
+    }
```

Comment:
I am sorry I didnt see this earlier. The duration given by the user is in seconds, while the internal times are all in milliseconds. we just need to make sure that value gets x 1000 somewhere